packages/aragon-wrapper/src/core/proxy/index.test.js

```diff
@@ -51,13 +51,119 @@ test('should get only the requested events', (t) => {
   }
   const instance = new Proxy(null, null, web3Stub)
   // act
-  const events = instance.events(['pay_fee', 'pay_service'])
+  const events = instance.events(['PayFee', 'PayService'])
   // assert
   events.subscribe(event => {
     t.deepEqual(event.amount, 5)
   })
 
-  eventEmitter.emit('data', { event: 'pay_fee', amount: 5 })
-  eventEmitter.emit('data', { event: 'pay_something_else', amount: 10 })
-  eventEmitter.emit('data', { event: 'pay_service', amount: 5 })
+  eventEmitter.emit('data', { event: 'PayFee', amount: 5 })
+  eventEmitter.emit('data', { event: 'PaySomethingElse', amount: 10 })
+  eventEmitter.emit('data', { event: 'PayService', amount: 5 })
+})
+
+test('should get only request the single event', (t) => {
+  t.plan(4)
+  // arrange
+  const allEventEmitter = new EventEmitter()
+  const payFeeEventEmitter = new EventEmitter()
+  const allEventsStub = sinon.stub().returns(allEventEmitter)
+  const payFeeEventStub = sinon.stub().returns(payFeeEventEmitter)
+  const contract = {
+    events: {
+      allEvents: allEventsStub,
+      PayFee: payFeeEventStub
+    }
+  }
+
+  const web3Stub = {
+    eth: {
+      Contract: sinon.stub().returns(contract)
+    }
+  }
+  const instance = new Proxy(null, null, web3Stub)
+  // act
+  const events = instance.events('PayFee')
+  // assert
+  t.true(payFeeEventStub.calledOnce)
+  t.true(allEventsStub.notCalled)
+
+  let eventCount = 0
+  events.subscribe(event => {
+    if (eventCount === 0) {
+      t.deepEqual(event.amount, 5)
+    } else if (eventCount === 1) {
+      t.deepEqual(event.amount, 10)
+    } else {
+      // Should only see it twice
+      t.fail()
+    }
+    ++eventCount
```

Comment:
Not sure if there's a better way to write this type of test when multiple emissions are expected.